transport/server.cc

```diff
@@ -618,6 +618,11 @@ future<> cql_server::connection::process_request() {
                     f.length, mem_estimate, _server._max_request_size));
         }
 
+        if (_server._requests_serving > _server._config.max_concurrent_requests) {
+            return make_exception_future<>(
+                    exceptions::overloaded_exception(format("too many in-flight requests: {}", _server._requests_serving)));
```

Comment:
a nitpick: maybe don't break that into two lines.

Reply:
Won't this always be the same number -- the configured limit? If this error is user facing, maybe worth mentioning how this limit was determined, i.e. mention the configuration item name.

Reply:
Right now it is, but the idea is to eventually™ make the threshold more elastic and automatically adapted. Then, the message will still make sense and potentially mention different values.

Reply:
Right, makes sense.

Reply:
It might still be worth mentioning how the limit was configured, it might prevent unnecessary questions from users. Something like https://github.com/scylladb/scylla/blob/1adf2cc848694ce9d2558aaea60800887d64057a/flat_mutation_reader.cc#L132-L135

This need not block this patchset, can be done as a follow-up -- if you agree to it.

Reply:
Sure, good idea. By the way, it's definitely not a blocker for this patchset, since it was queued 3 hours ago (:

Reply:
Oh, didn't notice. :)